gloo/docs/user_guides/tls_setup.md

```diff
@@ -0,0 +1,317 @@
+To start, let's make sure the `petstore` application is deployed:
+
+```bash
+kubectl apply -f \
+https://raw.githubusercontent.com/solo-io/gloo/master/example/petstore/petstore.yaml
+```
+
+If we query the gloo upstreams we should see it:
+
+```bash
+glooctl get upstream default-petstore-8080
++-----------------------+------------+----------+-------------------------+
+|       UPSTREAM        |    TYPE    |  STATUS  |         DETAILS         |
++-----------------------+------------+----------+-------------------------+
+| default-petstore-8080 | Kubernetes | Accepted | svc name:      petstore |
+|                       |            |          | svc namespace: default  |
+|                       |            |          | port:          8080     |
+|                       |            |          | REST service:           |
+|                       |            |          | functions:              |
+|                       |            |          | - addPet                |
+|                       |            |          | - deletePet             |
+|                       |            |          | - findPetById           |
+|                       |            |          | - findPets              |
+|                       |            |          |                         |
++-----------------------+------------+----------+-------------------------+
+```
+
+Now let's create a route to the petstore like [we did in the basic routing tutorial](../basic_routing)
+
+```bash
+glooctl add route \
+    --path-exact /sample-route-1 \
+    --dest-name default-petstore-8080 \
+    --prefix-rewrite /api/pets
+```
+Since we didn't explicitly create a VirtualSerivce, adding this route will create a default VirtualService named `default`.
+
+```bash
+glooctl get virtualservice default -o yaml
+```
+
+```yaml
+---
+metadata:
+  name: default
+  namespace: gloo-system
+  resourceVersion: "21723"
+status:
+  reportedBy: gateway
+  state: Accepted
+  subresourceStatuses:
+    '*v1.Proxy gloo-system gateway-proxy':
+      reportedBy: gloo
+      state: Accepted
+virtualHost:
+  domains:
+  - '*'
+  name: gloo-system.default
+  routes:
+  - matcher:
+      exact: /sample-route-1
+    routeAction:
+      single:
+        upstream:
+          name: default-petstore-8080
+          namespace: gloo-system
+    routePlugins:
+      prefixRewrite:
+        prefixRewrite: /api/pets
+```
+
+If we want to query the service to verify routing is working, we can like this:
+
+```bash
+curl $(glooctl proxy url --port http)/sample-route-1
+[{"id":1,"name":"Dog","status":"available"},{"id":2,"name":"Cat","status":"pending"}]
+```
+
+Let's enable HTTPS by configuring TLS/SSL for our VirtualService.
+
+### Configuring TLS/SSL in a VirtualService
+
+Before we add the TLS/SSL configuration, let's create a private key and certificate to use in our VirtualService. Obviously if you have your own key/cert pair, you can use those instead of creating self-signed certs here.
```

Comment:
Is it worth discussing when to include Root CA or not? There was a lot of confusion on Root CA on community Slack, so feels like we should have more discussion.

Reply:
Okay I think it's worth a mention here that `glooctl` and `kubectl` offer different options and we don't use RootCA unless we are planning to set up mTLS with our downstream.